### ISSUE TITLE
Losed notes

### ISSUE THREAD

**Issue by kashinios** [chocoford/ExcalidrawZ#64]
**Describe the bug**
I took notes during the last week, and now when I open the app they are lost. By I see them in history. 

**To Reproduce**
1. Make notes and close the app
2. Open app 2+ days after
3. Notes not expected on canvas but available in history

**Desktop (please complete the following information):**
 - OS: macOS 15.0.1
 - Version(Build) 1.2.8 (33)

**Additional context**
- Installation source: App Store
- Are you logged into an iCloud account?:  No.

**Screenshots**
<img width="1512" alt="Screenshot 2024-12-26 at 12 18 56" src="https://github.com/user-attachments/assets/d9c5a84a-92d7-405c-96b1-b3e4c5f5ef62" />

**Comment by chocoford** [comment]
Did it cause any loss of your data?

**Comment by kashinios** [comment]
No, I only want to notify about this case because like this product

**Comment by chocoford** [comment]
Alright, I was just worried that your files might have been lost because of my mistake 😂. I’ll continue checking for bugs in the code. Thanks for your feedback!

**Comment by vs40** [comment]
It happened to me as well today :(
Created new drawing and worked on it, saved it as a local file later, but saved file appeared on very early stage of drawing and most of the drawing is gone.

History shows only initial created point and few very early stages. 

Version 1.2.8 (33) from App Store on macOS 15.2

And now I'm seeing this error(German), when drawing: "Die Daten konnten nicht gelesen werden, weil sie fehlen" - The data could not be read because it is missing.

**Comment by chocoford** [comment]
Hello @vs40.
I'm sorry to hear about the issues you're facing. Regarding your description, did you edit the content directly in the `.excalidraw` file, or did you open the main program and export the excalidraw file?

**Comment by vs40** [comment]
> Hello [@vs40](https://github.com/vs40). I'm sorry to hear about the issues you're facing. Regarding your description, did you edit the content directly in the `.excalidraw` file, or did you open the main program and export the excalidraw file?

I never edited the content of the file directly and always used the main program.

**Comment by chocoford** [comment]
Hello, @vs40. Thank you for the information. You've reminded me that I need to adjust the checkpoint mechanism. Currently, during the ongoing editing of a file, it only repeatedly overwrites the same file history. This can lead to situations like the one you encountered, where the content is completely overwritten and ends up being empty.

However, we still can't pinpoint the exact cause of this issue. Do you remember if you added any uncommon elements to the canvas? I initially suspect that there might be some compatibility problems with the data format in `excalidraw`.

**Comment by chocoford** [comment]
In fact, I did notice some oversights regarding the compatibility of the data formats. I'm sorry about that.🥲

**Comment by vs40** [comment]
> Do you remember if you added any uncommon elements to the canvas? I initially suspect that there might be some compatibility problems with the data format in `excalidraw`

I'm used only basic tools like text and arrows.
Only "uncommon" thing would be some German letters in text: üöäß.
